source/js/buyers-guide/search/history.js

```diff
@@ -117,7 +117,12 @@ export function performInitialHistoryReplace(
       });
   }
 
-  Utils.sortProductCards();
+  // sorting priorities change when search text is applied
+  if (searchParameter) {
+    Utils.sortFilteredProducts();
+  } else {
+    Utils.sortProductCards();
+  }
```

Comment:
Hi @fessehaye , 

I don't know the specs and codebase well enough. Could you help clarify why we have two different sorting functions? I imagine the sorting logic would be the same whether it's a full product list or a filtered one. The only difference would be which project list we apply the same sorting function on.

Reply:
So `sortFilteredProducts()` applies how products should be sorted from last years update https://github.com/mozilla/foundation.mozilla.org/issues/7476

Where `sortProductCards()` applies what sort is applied from the pni-category dropdown that was added this years update.

So the sorting logic is different whether it's a full product list or a filtered one. This is a little confusing so we added it for discussion for the next PNI meeting!

Reply:
Got it! Thanks for the explanation.